### PR TITLE
tranche dev doc

### PR DESCRIPTION
**Pull request by akiezun**
addresses #372 
@droazen please review

### CODE REVIEW

**Comment by akiezun** [comment]
fixes #372

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/Tranche.java [hunk 1]
```diff
@@ -218,17 +187,42 @@ public double getRunningMetric(int i) {
         return tranches;
     }
 
-    private static Tranche findTranche( final List<VariantDatum> data, final TruthSensitivityMetric metric, final double trancheThreshold, final VariantRecalibratorArgumentCollection.Mode model ) {
-        logger.debug(String.format("  Tranche threshold %.2f => selection metric threshold %.3f", trancheThreshold, metric.getThreshold(trancheThreshold)));
+    public static double getSensitivityThreshold(double tranche) {
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/Tranche.java]: Add javadoc for this method

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/Tranche.java [hunk 2]
```diff
@@ -159,54 +158,24 @@ private static int getOptionalInteger(Map<String,String> bindings, String key, i
         }
     }
 
-    @VisibleForTesting
-    static final class TruthSensitivityMetric {
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/Tranche.java]: Why did you choose to eliminate this abstraction? Do you think that passing around a raw array of doubles is better than passing around a Metric object?

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/Tranche.java [hunk 3]
```diff
@@ -218,17 +187,42 @@ public double getRunningMetric(int i) {
...
+            runningSensitivity[i] = 1 - nCalledAtTruth / (1.0 * nTrueSites);
+        }
+        return runningSensitivity;
+    }
+
+    /**
+     * Creates a tranche of variants that have the truth sensitivity better than the threshold.
+     * Returns null if no such tranche exists.
+     */
+    private static Tranche findTranche( final List<VariantDatum> data, final double[] runningSensitivity, final double trancheThreshold, final VariantRecalibratorArgumentCollection.Mode model ) {
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/Tranche.java]: Document these parameters

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/Tranche.java [hunk 4]
```diff
@@ -237,10 +231,12 @@ private static Tranche findTranche( final List<VariantDatum> data, final TruthSe
         return null;
     }
 
-    private static Tranche trancheOfVariants( final List<VariantDatum> data, int minI, double ts, final VariantRecalibratorArgumentCollection.Mode model ) {
+    /**
+     * Creates the tranche of variants with with minimal LOD given by {@param minLod}.
+     */
+    private static Tranche trancheOfVariants( final List<VariantDatum> data, final double minLod, final double ts, final VariantRecalibratorArgumentCollection.Mode model ) {
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/Tranche.java]: Document these parameters

**Comment by droazen** [comment]
I don't think this branch satisfies https://github.com/broadinstitute/hellbender/issues/372 at all. There are still methods with 4+ undocumented parameters, and you eliminated what seemed a useful abstraction (`TruthSensitivityMetric`) in favor of passing around a raw array of doubles.

**Comment by droazen** [comment]
Back to @akiezun

**Comment by akiezun** [comment]
closing for now

**Pull request closed without merging**